entity-framework/core/modeling/data-seeding.md

```diff
@@ -7,19 +7,74 @@ uid: core/modeling/data-seeding
 
-To add entities that have a relationship the foreign key values need to be specified. Frequently the foreign key properties are in shadow state, so to be able to set the values an anonymous class should be used:
+To add entities that have a relationship the foreign key values need to be specified:
 
 [!code-csharp[Main](../../../samples/core/DataSeeding/DataSeedingContext.cs?name=PostSeed)]
 
-Once entities have been added, it is recommended to use [migrations](xref:core/managing-schemas/migrations/index) to apply changes. 
+If the entity type has any properties in shadow state an anonymous class can be used to provide the values:
+
+[!code-csharp[Main](../../../samples/core/DataSeeding/DataSeedingContext.cs?name=AnonymousPostSeed)]
+
+Owned entity types can be seeded in a similar fashion:
+
+[!code-csharp[Main](../../../samples/core/DataSeeding/DataSeedingContext.cs?name=OwnedTypeSeed)]
+
+See the [full sample project](https://github.com/aspnet/EntityFramework.Docs/tree/master/samples/core/DataSeeding) for more context.
+
+Once the data has been added to the model, [migrations](xref:core/managing-schemas/migrations/index) should be used to apply the changes.
+
+> [!TIP]
+> If you need to apply migrations as part of an automated deployment you can [create a SQL script](xref:core/managing-schemas/migrations/index#generate-sql-scripts) that can be previewed before execution.
 
 Alternatively, you can use `context.Database.EnsureCreated()` to create a new database containing the seed data, for example for a test database or when using the in-memory provider. Note that if the database already exists, `EnsureCreated()` will neither update the schema nor the seed data in the database.
+
+This type of seed data is managed by migrations and the script to update the data that's already in the database needs to be generated without connecting to the database. This imposes some restrictions:
+* The primary key value needs to be specified even if it's usually generated by the database. It will be used to detect data changes between migrations.
+* Previously seeded data will be removed if the primary key is changed in any way.
+
+Therefore this feature is most useful for static data that's not expected to change outside of migrations and does not depend on anything else in the database, for example ZIP codes.
+
+If your scenario includes any of the following it is recommended to use custom initialization logic described in the next section:
+* Temporary data for testing
+* Data that depends on database state
+* Data that uses alternate keys as identity
+* Data that needs key values to be generated by the database
+* Data that requires custom transformation (that is not handled by [value conversions](xref:core/modeling/value-conversions)), such as some password hashing
+* Data that requires calls to external API, such as ASP.NET Core Identity roles and users creation
```

Comment:
Maybe the two above are just data that can be generated or computed only at runtime?

Reply:
This list has a lot of redundancy on purpose to make it more scannable

Reply:
It is somehow not working well for me :trollface:. Maybe it is just me, but I find lists of orthogonal items easier to understand and remember

Reply:
We can wait for user feedback on this